src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -0,0 +1,277 @@
+
+                if (!baseType.IsObjectType())
+                {
+                    return FindValidPrintMethod(baseType, ContainingType.DeclaringCompilation);
+                }
+
+                return null;
+            }
+
+#if DEBUG
+            static bool modifiersAreValid(DeclarationModifiers modifiers)
+            {
+                if ((modifiers & DeclarationModifiers.AccessibilityMask) != DeclarationModifiers.Private && 
+                    (modifiers & DeclarationModifiers.AccessibilityMask) != DeclarationModifiers.Protected)
+                {
+                    return false;
+                }
+
+                modifiers &= ~DeclarationModifiers.AccessibilityMask;
+
+                switch (modifiers)
+                {
+                    case DeclarationModifiers.None:
+                    case DeclarationModifiers.Override:
+                    case DeclarationModifiers.Virtual:
+                        return true;
+                    default:
+                        return false;
+                }
+            }
+#endif
+        }
+
+        protected override (TypeWithAnnotations ReturnType, ImmutableArray<ParameterSymbol> Parameters, bool IsVararg, ImmutableArray<TypeParameterConstraintClause> DeclaredConstraintsForOverrideOrImplementation) MakeParametersAndBindReturnType(DiagnosticBag diagnostics)
+        {
+            var compilation = DeclaringCompilation;
+            var location = ReturnTypeLocation;
+            return (ReturnType: TypeWithAnnotations.Create(Binder.GetSpecialType(compilation, SpecialType.System_Boolean, location, diagnostics)),
+                    Parameters: ImmutableArray.Create<ParameterSymbol>(
+                        new SourceSimpleParameterSymbol(owner: this,
+                            TypeWithAnnotations.Create(Binder.GetWellKnownType(compilation, WellKnownType.System_Text_StringBuilder, diagnostics, location), NullableAnnotation.NotAnnotated),
+                            ordinal: 0, RefKind.None, "builder", isDiscard: false, Locations)),
+                    IsVararg: false,
+                    DeclaredConstraintsForOverrideOrImplementation: ImmutableArray<TypeParameterConstraintClause>.Empty);
+        }
+
+        protected override int GetParameterCountFromSyntax() => 1;
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            var F = new SyntheticBoundNodeFactory(this, ContainingType.GetNonNullSyntaxNode(), compilationState, diagnostics);
+            try
+            {
+                ImmutableArray<Symbol> printableMembers = ContainingType.GetMembers()
+                    .WhereAsArray(m => m.DeclaredAccessibility == Accessibility.Public && (m.Kind == SymbolKind.Field || m.Kind == SymbolKind.Property));
+
+                if (printableMembers.Any((m, _) => SymbolExtensions.GetTypeOrReturnType(m).Type.IsErrorType(), arg: (object?)null))
```

Comment:
>, arg: (object?)null [](start = 109, length = 20)

There is an overload in `System.Linq.ImmutableArrayExtensions` that does not require an extra argument.

Reply:
Thanks. I have a mental block on importing `System.Linq` at this point, so couldn't find it ;-P

---
In reply to: [469463156](https://github.com/dotnet/roslyn/pull/46616#discussion_r469463156) [](ancestors = 469463156)